logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -101,10 +104,14 @@ private void addBody(final HttpMessage request, final ImmutableMap.Builder<Strin
         final String body = request.getBodyAsString();
 
         if (isJson(request.getContentType())) {
-            builder.put("body", body.isEmpty() ? JsonBody.EMPTY : new JsonBody(compactJson(body)));
-        } else {
-            addUnless(builder, "body", request.getBodyAsString(), String::isEmpty);
+            try {
+                builder.put("body", body.isEmpty() ? JsonBody.EMPTY : new JsonBody(compactJson(body)));
+                return;
```

Comment:
> Yes - actually, why do we differ in behavior for empty JSON-body and empty non-JSON-body @whiskeysierra ?

I think we can switch those conditions and 

1. check for empty body
2. check for content type json